src/main/java/com/google/devtools/build/lib/bazel/bzlmod/Version.java

```diff
@@ -51,6 +51,13 @@
  * {@code a} and {@code b} compare {@code a < b} iff the same holds when they're compared as Bazel
  * module versions.
  *
+ * <p>Versions with a "build" part are generally accepted as input, but they're treated as if the
```

Comment:
We should enforce this in the BCR by preventing "equal" but distinct versions.

Reply:
it's "effectively" enforced right now because:

1) up until this CL, `+` is an invalid character in repo names, so nobody could have used a version with a `+` sign in it without crashing Bazel.
2) after this CL, when Bazel sees a `bazel_dep(name='foo',version='1.0.0+build')`, it'll request `%registry%/modules/foo/1.0.0`. So the `+build` part will always be stripped, and a submission to BCR that has version `1.0.0+build` is unusable.

but we can make the check explicit. just not very high priority.